### ISSUE TITLE
Flow / Chatbot / Notify support

### ISSUE THREAD

**Issue by gymnae** [pulsejet/memories#651]
**Is your feature request related to a problem? Please describe.**
I use automated tagging to assign tags to certain folders. A flow is then triggering a bot to write about that into a NC talk chat. It's linking to the file directly, which has its limitations and requires me to tell people to please look at the link I provided to access the gallery.
Without automated tagging, it would still be nice to have an memories tuned method to use the notify mechanics of Nextcloud to create links to media consumption in memories

**Describe the solution you'd like**
That memories offers a flow to notify via email / notifier / chatbot about new items in a chosen album or folder – Linking directly to the _memories_ view of these items

**Describe alternatives you've considered**
Described above, using the current bots to show that something new is there, but telling people to not click on the items from the chatbot

**Comment by pulsejet** [comment]
I've no clue how flow works, but wouldn't this be part of a separate app? Also, there's the issue of permissions and timeline paths. Are the created links public shares? Or everyone is using a common nextcloud user?

**Comment by Gymnae** [comment]
The chat is only visible to Nextcloud users, all of which have access to the folders containing the shared images. Permission wise it would not be a problem. 

In general, making use of the activity/notification backends in order to consume custom memories events downstream could be nice. Users could create custom flows, depending on the actions of memories, for example based on activity when creating shares or adding images to albums.

I'm considering doing a regex replace via the reverse proxy, to add /apps/memories/ to every share link, but I cannot determine the share type and it would break sharing non images/videos.

**Comment by Gymnae** [comment]
This would be the flow from the talk app which could serve as a base, but I'm no coder to get really further: 
https://github.com/nextcloud/spreed/blob/master/lib/Flow/RegisterOperationsListener.php

**Comment by gymnae** [comment]
There's news regarding this feature request.
Nextcloud Talk now offers an integrated way to add Bots:
https://nextcloud-talk.readthedocs.io/en/latest/bots/

**Comment by pulsejet** [comment]
Yeah that sounds very cool. Some of these events probably need to be implemented in either Photos or a separate app though. For instance Memories just uses the DAV API of Photos to add files to albums, so we never see that happening.
